integration-playbook: Add hidden dependencies

cevich opened the pull request:
Previously, an internal playbook installed many extra
necessary/unnecessary packages before this playbook even started.  Since
this is a terrible design, move all dependencies here so that nothing is
unwritten.  This includes installing some deps. for ansible itself
(which must be done as a raw command).

Signed-off-by: Chris Evich <cevich@redhat.com>

cevich commented:
CI run started manually: https://hurl.corp.redhat.com/8c6d900

cevich commented:
Hmm, closer, maybe another missing package is causing the 0/0 failure.  Will need to look deeper, but this PR certainly made the situation better (reports + RHEL & Fedora completing)

mrunalp commented on contrib/test/crio-integration-playbook.yaml:
```diff
@@ -57,6 +57,7 @@
       name: "{{ item }}"
       state: latest
     with_items:
+      - iptables
```
I am curious. Did we change the image so we now have to install iptables?

cevich replied:
It was part of that "hidden" playbook, which is a terrible place to put dependencies.  They all need to be spelled out here, otherwise the tests are beholden to some external entity "getting it right".  In the interest of time, I can copy paste all of them here, then y'all can weed out the ones you don't need, later.

mrunalp replied:
okay

mrunalp commented:
Also, I see nc missing in 1602. PTAL at the RHEL fullresult.txt.

cevich commented:
Started over...move ALL the dependencies setup in by the "hidden" playbook to here.  Now that playbook is no-longer needed and this one should be able to stand on it's own.  I "lightly" tested these changes by hand.  Let's see what Jankenz thinks...

cevich commented:
CI Run started manually: https://hurl.corp.redhat.com/040af23

cevich commented:
```
08:42:47     import paramiko
08:42:47 ImportError: No module named paramiko
08:42:47 ERROR: Provisioning\nSTATUS: 1
```

Right...that package was on the slave last night.  To me, it appears like someone changed the the Jenkins slave out from under us 😠

cevich commented:
...aye, tried again.  paramiko is a dependency of OLD ansible (used on the slave).  That was working fine last night and I haven't touched the slave.  Digging...

cevich commented:
(rebased / force-pushed)

cevich commented:
(rebased / force-pushed)

The 'Run integration tests on XYZ' tasks are now chooching and no tasks (up to that point) fail because of missing dependencies.

**N/B:** (this means you @stevekuznetsov) The playbook now requires passing in two variables (and it makes sure you don't forget):  ``extras_repo_baseurl`` and ``server_repo_baseurl``.  Their values should be obvious based on their names.  Let me know if there's a more betterer / preferred way to deal with this.

cevich commented:
@mrunalp Great news:

```
PLAY RECAP **********************************************************************************************
f26_pr_928_workspace_0     : ok=66   changed=42   unreachable=0    failed=0
rhel74_pr_928_workspace_0  : ok=70   changed=57   unreachable=0    failed=0
```

Mailing you the results.

TomSweeneyRedHat commented:
LGTM, would like to get a head nod from  @stevekuznetsov

cevich commented:
Actually we're missing one more thing that's really important here...

cevich commented:
...because not having the results is just as good as failing 😆

I tested the new role, tar'd up the results directory and e-mailed it to y'all.

@stevekuznetsov I've got to do another run of this anyway, because 99% sure isn't good enough (I added the 'nc' package and 'crio_results' role).  Let's coordinate WRT  #927 to resolve conflicts.  I'm "okay" with include files but would prefer to go straight to very-basic roles, even if they don't use variables.  There's a trello-card @runcom has pending that will be much easier to complete with roles instead of includes.

cevich commented:
(typo fix)

cevich commented:
Uggg, just when you think you've got it, something else breaks.  Anyone see this before:

```
TASK [Set custom Golang path for Fedora/RHEL] ***********************************************************
changed: [f26_pr_928_workspace_2]
changed: [rhel74_pr_928_workspace_2]

TASK [Copy redhat sigstore file to test host] ***********************************************************
changed: [f26_pr_928_workspace_2]
fatal: [rhel74_pr_928_workspace_2]: FAILED! => {"changed": false, "failed": true, "msg": "Destination directory /etc/containers/registries.d does not exist"}

TASK [Copy test policy.json file] ***********************************************************************
changed: [f26_pr_928_workspace_2]
```

I added a fix above (23e3f9c) but it feels wrong since presumably that was not required prior to this PR.  Perhaps is it a sign RHEL is still missing some package?

cevich commented:
All, nvm, @rhatdan found the problem.  Fix is above.  Test is still choochin'

TomSweeneyRedHat commented:
LGTM assuming happy tests and happy  @stevekuznetsov

cevich commented:
Yes, please @stevekuznetsov double-check.  I'm known to make typos and deploy logic-bafoonery with some frequency 😀

cevich commented:
(my manual run is getting ready to execute 'k8s conformance tests'), all VMs are happy so far.)

cevich commented:
okay, we're 100% pass test-wise.  However I made manual updates to account for the --fixups here.  Squashing those away and removing the 'results collection' part (extra uncertainty).  Starting a squeeky-clean run...

@stevekuznetsov for your #927 (if you don't mind) we need an addition similar to: af6d1f0d879dfa8d2aa5e6ae6dff4dd47870e399 (not working).  No worries if that's too much, we can do it in a separate PR.  It just saves some handy-work collecting results.

cevich commented on contrib/test/crio-integration-playbook.yaml:
```diff
@@ -1,12 +1,67 @@
-## --skip-tags pr
+## --skip-tags pr.
+##
+## Required: --extra-vars extras_repo_baseurl=<HTTP or HTTPS>
```
Uggg, that should also include mention of ``server_repo_baseurl`` being required.  Not fixing it in the interest of getting a run going quickly...

cevich commented on contrib/test/crio-integration-playbook.yaml:
```diff
@@ -27,28 +82,57 @@
+      - rpcbind
+      - rsync
+      - sed
       - socat
```
There are probably a few deps. above that are not strictly needed.  Leaving that for someone else to fix 😀

cevich commented:
@mrunalp @rhatdan @TomSweeneyRedHat we finally got a golden PASS.

**N/B:** This is after doing 3-runs of this PR without anything else changing, plus 1 run prior to rebase/squash.  For 3-out-of-4 runs, we had failures of **different** tests each time.  This suggests there are some tests dependent on race-conditions.  If this PR is merged, I expect to see this same behavior on all PR testing until those problems are found/addressed.

```
[root@cevich-testing-fedora pr_928_workspace_6]# tail results/*/*
==> results/f26_pr_928_workspace_6/node-e2e.log <==
  /root/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/common/configmap.go:38
------------------------------
SSSSSSSSSSep 21 19:27:52.341: INFO: Running AfterSuite actions on all node
Sep 21 19:27:52.341: INFO: Running AfterSuite actions on node 1

Ran 149 of 652 Specs in 3665.263 seconds
SUCCESS! -- 149 Passed | 0 Failed | 0 Pending | 503 Skipped PASS

Ginkgo ran 1 suite in 1h1m5.513778265s
Test Suite Passed

==> results/f26_pr_928_workspace_6/testout.txt <==
ok 149 ctr seccomp profiles falls back to pod profile
ok 150 ctr seccomp profiles falls back to unconfined
ok 151 pod seccomp profiles unconfined
ok 152 pod seccomp profiles runtime/default
ok 153 pod seccomp profiles wrong profile name
ok 154 # skip (need https://issues.k8s.io/36997) pod seccomp profiles localhost/profile_name
ok 155 ctr termination reason Completed
ok 156 kpod tag with shortname:latest
ok 157 kpod tag with shortname
ok 158 kpod tag with shortname:tag

==> results/rhel74_pr_928_workspace_6/node-e2e.log <==
  /root/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/common/configmap.go:38
------------------------------
SSSSSSSSSSep 21 19:19:39.563: INFO: Running AfterSuite actions on all node
Sep 21 19:19:39.563: INFO: Running AfterSuite actions on node 1

Ran 149 of 652 Specs in 3172.835 seconds
SUCCESS! -- 149 Passed | 0 Failed | 0 Pending | 503 Skipped PASS

Ginkgo ran 1 suite in 52m53.111217493s
Test Suite Passed

==> results/rhel74_pr_928_workspace_6/testout.txt <==
ok 149 ctr seccomp profiles falls back to pod profile
ok 150 ctr seccomp profiles falls back to unconfined
ok 151 pod seccomp profiles unconfined
ok 152 pod seccomp profiles runtime/default
ok 153 pod seccomp profiles wrong profile name
ok 154 # skip (need https://issues.k8s.io/36997) pod seccomp profiles localhost/profile_name
ok 155 ctr termination reason Completed
ok 156 kpod tag with shortname:latest
ok 157 kpod tag with shortname
ok 158 kpod tag with shortname:tag
[root@cevich-testing-fedora pr_928_workspace_6]# grep 'not ok' results/*/*
[root@cevich-testing-fedora pr_928_workspace_6]# grep -c 'ok ' results/*/testout.txt
results/f26_pr_928_workspace_6/testout.txt:158
results/rhel74_pr_928_workspace_6/testout.txt:158
```

mrunalp commented:
LGTM
